mslib/msui/wms_control.py

```diff
@@ -660,9 +664,15 @@ def get_capabilities(self):
         else:
             # url shortener url translated
             url = request.url
-            url = url.replace('?service=WMS&request=GetCapabilities&version=1.1.1', '')
+
+            # Take the default version of the server, or 1.3.0 if not supported
+            version = request.text.split("MS_Capabilities version=\"")[-1].split("\"")[0]
+            if version not in ["1.1.1", "1.3.0"]:
+                version = "1.3.0"
```

Comment:
Instead of hardcoding the version for the client, the client scrapes the version from the default GetCapabilities document of the server.
Is this a good or a bad approach?

Reply:
The old approach seems very inflexible/wrong.
But the new one...
Isn't there a standard way to parse the URL for the parameters and use that?

Reply:
I changed it from scraping the response, to parsing the response and checking the version through that.
This should be more robust now.